include/nathelpers/typeinfo.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+
+namespace Natalie {
+
+template <class T>
+struct typeinfo {
+    constexpr const char *name() = delete;
```

Comment:
These are all done at compile time. When asked for a value that is not yet defined, it results in a compilation error that's easy to solve by adding a more specific `typeinfo` struct down below.